atdgen-runtime/src/json_adapter.ml

```diff
@@ -19,7 +19,7 @@ module Type_field = struct
       match x with
       | `Assoc fields ->
           (match List.assoc type_field_name fields with
-           | `String type_ -> `List [ `String type_; x ]
+           | `String _type_ -> `List [ `String _type_; `Assoc (List.remove_assoc type_field_name fields) ]
```

Comment:
The leading underscore is reserved for intentionally unused identifiers. It shouldn't be used here.